File: leetcode/container_most_water/ContainerMostWater.cpp
class Solution {
public:
    int maxArea(vector<int> &height) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        int i = 0; 
        int j = height.size() - 1; 
        int maxArea = 0; 
        
        while (i <= j) {
            int currArea = min(height[i], height[j]) * (j - i); 
            
            if (currArea > maxArea) {
                maxArea = currArea;  
            }
            
            if (height[i] <= height[j]) {
                i++; 
            }
            else {
                j--; 
            }
        }
        
        return maxArea; 
    }
};

File: leetcode4/ContainerMostWater.java
public class Solution {
    public int maxArea(int[] height) {
        int len = height.length; 
        
        if (len == 0) {
            return 0; 
        }
        
        int left = 0; 
        int right = len - 1; 
        int max = (right - left) * Math.min(height[left], height[right]);
        
        while (left < right) {
            if (height[left] < height[right]) {
                left++; 
            } else {
                right--; 
            }
            
            max = Math.max(max, (right - left) * Math.min(height[left], height[right])); 
        }
        
        return max; 
    }
}
